eos/utils/kmatrix-impl.hh

```diff
@@ -133,33 +137,51 @@ namespace eos
     }
 
 
-    // Return the row corresponding to the index rowindex of the T matrix defined as T = (1-i*rho*K)^(-1)*K
+    // Return the row corresponding to the index rowindex of the T matrix defined as T = n * (1 - i * K * rho * n * n)^(-1) * K * n
     template <unsigned nchannels_, unsigned nresonances_>
     std::array<complex<double>, nchannels_>
-    KMatrix<nchannels_, nresonances_>::tmatrix_row(unsigned rowindex, double _s) const
+    KMatrix<nchannels_, nresonances_>::tmatrix_row(unsigned rowindex, complex<double> _s) const
     {
         std::array<complex<double>, nchannels_> tmatrixrow;
+        std::array<complex<double>, nchannels_> nfactors;
         const auto & channels = this->_channels;
         const auto & resonances = this->_resonances;
         const auto & bkgcst = this->_bkgcst;
         // Adapt s to avoid pole in the K matrix
-        const double s = adapt_s(_s);
+        const complex<double> s = adapt_s(_s);
 
-        ///////////////////
-        //1. Fill tmp1 = rho matrix
-        ///////////////////
         gsl_matrix_complex_set_zero(_tmp_1);
         gsl_matrix_complex_set_zero(_tmp_2);
 
         for (size_t i = 0 ; i < nchannels_ ; ++i)
         {
-            complex<double> rhoentry = channels[i]->rho(s);
-            gsl_matrix_complex_set(_tmp_1,  i,  i, gsl_complex_rect(rhoentry.real(), rhoentry.imag()));
+            const unsigned li = channels[i]->_l_orbital;
+            const double q0 = channels[i]->_q0.evaluate();
+            const complex<double> mi1_2 = power_of<2>((double)channels[i]->_m1);
```

Comment:
Why are you explicitly casting this to double? Is it a parameter? If yes, then it has a function call operator ``()``.